app/src/main/java/uk/gov/onelogin/network/auth/AuthCodeExchange.kt

```diff
@@ -0,0 +1,91 @@
+package uk.gov.onelogin.network.auth
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.VisibleForTesting.Companion.PRIVATE
+import com.google.gson.Gson
+import dagger.hilt.android.qualifiers.ApplicationContext
+import io.ktor.client.request.post
+import io.ktor.client.statement.bodyAsText
+import io.ktor.http.ContentType
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.Url
+import io.ktor.http.contentType
+import uk.gov.onelogin.R
+import uk.gov.onelogin.network.auth.response.TokenResponse
+import uk.gov.onelogin.network.http.IHttpClient
+import uk.gov.onelogin.network.utils.IOnlineChecker
+import javax.inject.Inject
+
+class AuthCodeExchange @Inject constructor(
+    @ApplicationContext
+    private val context: Context,
+    private val httpClient: IHttpClient,
+    private val onlineChecker: IOnlineChecker
+): IAuthCodeExchange {
```

Comment:
The constructor arguments are injected for us by [Hilt](https://dagger.dev/hilt/)